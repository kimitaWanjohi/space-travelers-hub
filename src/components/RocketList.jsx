import { useSelector, useDispatch } from "react-redux";
import { useEffect } from "react";
import { fetchRockets } from "../redux/rockets/rocketSlice";
import Rocket from "./Rocket";

const RocketList = () => {
  const { rockets, isLoading, error } = useSelector((store) => store.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchRockets());
  }, [dispatch]);

  return (
    <div className="mt-6 mb-8">
      {isLoading && <div>Loading...</div>}
      {error && <div>{error}</div>}
      {rockets && (
        <ul>
          {rockets.map((rocket) => (
            <Rocket
              key={rocket.id}
              id={rocket.id}
              name={rocket.name}
              description={rocket.description}
              images={rocket.flickr_images[0]}
              reserved={rocket.reserved}
            />
          ))}
        </ul>
      )}
    </div>
  );
};

export default RocketList;
