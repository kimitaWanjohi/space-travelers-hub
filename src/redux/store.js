import { configureStore } from "@reduxjs/toolkit";
import { createLogger } from "redux-logger";
import missionsReducer from "./missions/slice";

export default configureStore({
  reducer: {
    missions: missionsReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(createLogger()),
});
