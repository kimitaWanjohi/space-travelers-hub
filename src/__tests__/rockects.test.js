import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from '../redux/store';
import RocketList from "../components/RocketList";

describe("Missions", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <RocketList />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders a list of missions", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <RocketList />
        </Provider>
      )
      .toJSON();
    expect(tree.children.length).toBe(2);
  });
});
