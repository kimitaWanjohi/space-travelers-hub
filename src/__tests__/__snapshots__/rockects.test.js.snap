// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions renders correctly 1`] = `
<div
  className="mt-6 mb-8"
>
  <div>
    Loading...
  </div>
  <ul />
</div>
`;
